### PR TITLE
Formula for building and initializing Hashicorp Vault

### PR DESCRIPTION
**Pull request by blarghmatey**
Created a set of states to install, configure, and initialize a server with the Hashicorp Vault project for managing sensitive data.

### CODE REVIEW

File: minion.conf [hunk 1]
```diff
@@ -6,8 +6,8 @@ fileserver_backend:
 
 gitfs_provider: gitpython
 
-
 gitfs_remotes:
   - https://github.com/mitodl/salt-extensions:
       - root: extensions
 
+vault.verify: False
```
Comment (bdero) [inline minion.conf]: Nit: Add newline

File: pillar.example [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
-    conf_file: /usr/local/etc/vault/vault.conf
+  overrides:
+    keybase_users:
+      - renaissancedev
+      - pdpinch
+      - bdero
+    secret_shares: 3
+    secret_threshold: 2
   config:
     key: othervalue
```
Comment (bdero) [inline pillar.example]: It doesn't look like this "config" key in the pillar is referenced. Is it being used?

File: vault/initialize.sls [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    - name: git+https://github.com/mitodl/hvac
+    - reload_modules: True
+
+initialize_vault_server:
+  vault.initialized:
+    - secret_shares: {{ vault.secret_shares }}
+    - secret_threshold: {{ vault.secret_threshold }}
+    - unseal: {{ vault.unseal }}
+    - pgp_keys: {{ vault.pgp_keys }}
+    - keybase_users: {{ vault.keybase_users }}
```
Comment (bdero) [inline vault/initialize.sls:17]: Woah, keybase support! How exactly will this be used?

File: pillar.example [hunk 4]
```diff
@@ -1,6 +1,10 @@
...
-    key: othervalue
+  overrides:
+    keybase_users:
+      - renaissancedev
+      - pdpinch
+      - bdero
+    secret_shares: 3
+    secret_threshold: 2
+
+vault.verify: False
```
Comment (bdero) [inline pillar.example]: Is vault.verify used for something? I can't find it referenced in the rest of this PR.
Also, I noticed there's a grain defined for this. Do we need `vault.verify` to be both pillar data and a grain?

**Comment by bdero** [comment]
This looks good to me after my one new comment. :+1:

File: pillar.example (hunk 4 above)
Comment (blarghmatey) [inline pillar.example]: The vault.verify setting is picked up by the execution module to specify whether to validate the TLS certificate used by Vault.

**Pull request merged**